File: server/src/main/java/com/smartexpense/api/domain/repository/UserRepository.java
package com.smartexpense.api.domain.repository;

import com.smartexpense.api.domain.model.User;

import java.util.Optional;
import java.util.UUID;

public interface UserRepository {
    Optional<User> findById(UUID id);
    Optional<User> findByEmail(String email);
    boolean existsByEmail(String email);
    User save(User user);
}

File: server/src/main/kotlin/com/smartexpense/api/domain/repository/UserRepository.kt
package com.smartexpense.api.domain.repository

import com.smartexpense.api.domain.model.User
import java.util.UUID

interface UserRepository {
    fun findById(id: UUID): User?
    fun findByEmail(email: String): User?
    fun existsByEmail(email: String): Boolean
    fun save(user: User): User
}
